bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/DreamScreenHandler.java

```diff
@@ -0,0 +1,240 @@
+            case CHANNEL_INPUT:
+                if (command instanceof DecimalType) {
+                    changeInput((DecimalType) command);
+                }
+                break;
+        }
+    }
+
+    private void changePower(OnOffType command) {
+        try {
+            send(0x03, 0x01, new byte[] { command == ON ? powerOnMode.deviceMode : 0 });
+            this.powerOn = command == ON;
+            updateState(CHANNEL_POWER, command);
+        } catch (IOException e) {
+            logger.debug("Error changing power state", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Cannot send power command to " + getName());
+        }
+    }
+
+    private void changeMode(DecimalType state) {
+        try {
+            final DreamScreenEnumMode mode = DreamScreenEnumMode.fromState(state);
+            if (this.powerOn) {
+                send(0x03, 0x01, new byte[] { mode.deviceMode });
+            }
+            this.powerOnMode = mode;
+            updateState(CHANNEL_MODE, state);
+        } catch (IOException e) {
+            logger.debug("Error changing mode", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Cannot send mode command to " + getName());
+        }
+    }
+
+    private void changeScene(DecimalType state) {
+        try {
+            final DreamScreenEnumScene scene = DreamScreenEnumScene.fromState(state);
+            send(0x03, 0x08, new byte[] { scene.deviceAmbientSceneType });
+            if (scene.deviceAmbientSceneType == 1) {
+                send(0x03, 0x0D, new byte[] { scene.deviceAmbientScene });
+            }
+            updateState(CHANNEL_SCENE, state);
+        } catch (IOException e) {
+            logger.debug("Error changing scene", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Cannot send scene command to " + getName());
+        }
+    }
+
+    private void changeInput(DecimalType state) {
+        try {
+            send(0x03, 0x20, new byte[] { state.byteValue() });
+            updateState(CHANNEL_INPUT, state);
+        } catch (IOException e) {
+            logger.debug("Error changing input", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Cannot send input change command to " + getName());
+        }
+    }
+
+    private void send(int commandUpper, int commandLower, byte[] payload) throws IOException {
+        if (this.address != null) {
+            server.send(this.group, commandUpper, commandLower, payload, this.address);
+        } else {
+            logger.warn("DreamScreen {} is not on-line", getName());
+        }
+    }
+
+    void refreshState(final byte[] data, int off, int len, InetAddress address) {
+        this.address = address;
+        this.group = data[off + 38];
+
+        logger.trace("{}", data.toString());
```

Comment:
You don't need to use `toString`:
```suggestion
        logger.trace("{}", data);
```

Reply:
changed